src/sdl2_compat.c

```diff
@@ -8211,12 +8209,15 @@ SDL_CreateWindow(const char *title, int x, int y, int w, int h, Uint32 flags)
 {
     SDL_Window *window = NULL;
     const Uint32 is_popup = flags & (SDL_WINDOW_POPUP_MENU | SDL_WINDOW_TOOLTIP);
+    bool exclusive_fullscreen = false;
 
     CheckEventFilter();
 
-    if (flags & SDL2_WINDOW_FULLSCREEN_DESKTOP) {
+    if ((flags & SDL2_WINDOW_FULLSCREEN_DESKTOP) == SDL2_WINDOW_FULLSCREEN_DESKTOP) {
         flags &= ~SDL2_WINDOW_FULLSCREEN_DESKTOP;
         flags |= SDL_WINDOW_FULLSCREEN; /* This is fullscreen desktop for new windows */
+    } else if (flags & SDL_WINDOW_FULLSCREEN) {
+        exclusive_fullscreen = true;
```

Comment:
I think we also need to unset the `SDL_WINDOW_FULLSCREEN` here to get the correct mode selected. When I was testing this commit creating a 640x480 fullscreen window, the resulting fullscreen mode ended up being the desktop mode. When SDL3 enters fullscreen desktop mode during window creation, the call to `SDL3_GetWindowSize()` will return the desktop dimensions and cause us to switch to the wrong mode.

Reply:
Oh, good point. We should probably also mark the window hidden if we’re about to set a full screen mode.